lib/services/ngsi/ngsiService.js

```diff
@@ -584,14 +741,172 @@ function sendUpdateValueNgsi2(entityName, attributes, typeInformation, token, ca
+                    delete result.type;
+                    options.json = result;
+                    logger.debug(context, 'typeInformation: %j', typeInformation);
+                    if ( ('timestamp' in typeInformation && typeInformation.timestamp !==
+                        undefined) ? typeInformation.timestamp : config.getConfig().timestamp) {
+                        if (!utils.isTimestampedNgsi2(options.json)) {
+                            options.json = addTimestampNgsi2(options.json, typeInformation.timezone);
+                        } else if (!utils.IsValidTimestampedNgsi2(options.json)) {
+                            logger.error(context, 'Invalid timestamp:%s', JSON.stringify(options.json));
+                            callback(new errors.BadTimestamp(options.json));
+                            return;
+                        }
+                    }
+                }
+            } else {
+                delete payload.id;
+                delete payload.type;
+                options.json = payload;
+            }
+            // Purge object_id from entities before sent to CB
+            // object_id was added by createNgsi2Entity to allow multientity
+            // with duplicate attribute names.
+            var att;
+            if (options.json.entities) {
+                for (var entity = 0; entity < options.json.entities.length; entity++) {
+                    for (att in options.json.entities[entity]) {
+                        /*jshint camelcase: false */
+                        if (options.json.entities[entity][att].object_id) {
+                            /*jshint camelcase: false */
+                            delete options.json.entities[entity][att].object_id;
+                        }
+                        if (options.json.entities[entity][att].multi) {
+                            delete options.json.entities[entity][att].multi;
+                        }
+                    }
+                }
+            } else {
+                for (att in options.json) {
+                    /*jshint camelcase: false */
+                    if (options.json[att].object_id) {
+                        /*jshint camelcase: false */
+                        delete options.json[att].object_id;
+                    }
+                    if (options.json[att].multi) {
+                        delete options.json[att].multi;
+                    }
+                }
+            }
+
+            options.json = formatAsNGSILD(options.json);
+            options.method = 'PATCH';
+
+            logger.debug(context, 'Updating device value in the Context Broker at [%s]', options.url);
+            logger.debug(context, 'Using the following NGSI-LD request:\n\n%s\n\n', JSON.stringify(options, null, 4));
+
+            request(options,
+                generateNGSILDOperationHandler('update', entityName, typeInformation, token, options, callback));
+        }
+    });
+}
+
+
+
+
+
+
+
+
+
```

Comment:
Big bunch of empty blank lines... intentional?

Reply:
Fixed b41a18a  (if only this tidying could be done automatically)